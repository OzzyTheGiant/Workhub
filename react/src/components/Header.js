import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => {
    return (
        <header id="Header">
            <img src="http://placehold.it/120x35" alt="Workhub Logo"/>
            <nav>
                <ul>
                    <li onClick={props.logout}>Log Out</li>
                </ul>
            </nav>
        </header>
    );
};

Header.propTypes = {
    logout:PropTypes.func.isRequired
}

export default Header;