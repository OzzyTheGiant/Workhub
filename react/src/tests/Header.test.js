import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Header from 'components/Header';

describe("Header", () => {
	/* ==== COMPONENT ==== */
	const header = <Header/>

	beforeAll(() => {
		configure({adapter:new Adapter()})
	});

	it("Should render correctly", () => {
		const componentWrapper = mount(header);
		let jsonComponent = toJson(componentWrapper);
   		expect(jsonComponent.type).toBe('Header');
   		expect(jsonComponent.children[0].type).toBe('header');
        expect(jsonComponent).toMatchSnapshot();
	});
});
