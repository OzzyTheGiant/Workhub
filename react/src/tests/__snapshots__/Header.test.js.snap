// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly 1`] = `
<Header>
  <header
    id="Header"
  >
    <img
      alt="Workhub Logo"
      src="http://placehold.it/120x35"
    />
    <nav>
      <ul>
        <li>
          Documents
        </li>
        <li>
          Log Out
        </li>
      </ul>
    </nav>
  </header>
</Header>
`;
